src/tables/colr.rs

```diff
@@ -91,43 +751,1153 @@ impl<'a> Table<'a> {
+                    painter,
+                    recursion_stack,
+                    #[cfg(feature = "variable-fonts")]
+                    coords,
+                    foreground_color,
+                );
+                painter.push_layer(composite_mode);
+                self.parse_paint(
+                    offset + source_paint_offset.to_usize(),
+                    palette,
+                    painter,
+                    recursion_stack,
+                    #[cfg(feature = "variable-fonts")]
+                    coords,
+                    foreground_color,
+                );
+                painter.pop_layer();
+                painter.pop_layer();
             }
+            _ => {}
         }
 
         Some(())
     }
+
+    fn parse_color_line(
+        &self,
+        offset: usize,
+        foreground_color: RgbaColor,
+    ) -> Option<NonVarColorLine<'a>> {
+        let mut s = Stream::new_at(self.data, offset)?;
+        let extend = s.read::<GradientExtend>()?;
+        let count = s.read::<u16>()?;
+        let colors = s.read_array16::<ColorStopRaw>(count)?;
+        Some(NonVarColorLine {
+            extend,
+            colors,
+            foreground_color,
+            palettes: self.palettes,
+        })
+    }
+
+    #[cfg(feature = "variable-fonts")]
+    fn parse_var_color_line(
+        &self,
+        offset: usize,
+        foreground_color: RgbaColor,
+    ) -> Option<VarColorLine<'a>> {
+        let mut s = Stream::new_at(self.data, offset)?;
+        let extend = s.read::<GradientExtend>()?;
+        let count = s.read::<u16>()?;
+        let colors = s.read_array16::<VarColorStopRaw>(count)?;
+        Some(VarColorLine {
+            extend,
+            colors,
+            foreground_color,
+            palettes: self.palettes,
+        })
+    }
+}
+
+struct RecursionStack {
+    // The limit of 64 is chosen arbitrarily and not from the spec. But we have to stop somewhere...
+    stack: [usize; 64],
+    len: usize,
+}
+
+impl RecursionStack {
+    #[inline]
+    pub fn is_empty(&self) -> bool {
+        self.len == 0
+    }
+
+    #[inline]
+    pub fn push(&mut self, offset: usize) -> Result<(), ()> {
+        if self.len == self.stack.len() {
+            Err(())
+        } else {
+            self.stack[self.len] = offset;
+            self.len += 1;
+            Ok(())
+        }
+    }
+
+    #[inline]
+    pub fn contains(&self, offset: usize) -> bool {
+        if let Some(offsets) = self.stack.get(..self.len) {
+            return offsets.contains(&offset);
+        }
+
+        false
+    }
+
+    #[inline]
+    pub fn pop(&mut self) {
+        debug_assert!(!self.is_empty());
+        self.len -= 1;
+    }
+}
+
+#[cfg(feature = "variable-fonts")]
+#[derive(Clone, Copy, Debug, Default)]
+struct VariationData<'a> {
+    variation_store: Option<ItemVariationStore<'a>>,
+    delta_map: Option<DeltaSetIndexMap<'a>>,
+}
+
+#[cfg(feature = "variable-fonts")]
+impl VariationData<'_> {
```

Comment:
No way we can avoid const generics?
I haven't familiarized myself with them yet. This is not about MSRV, but more about maintenance.

Reply:
Sure you can avoid it, but then you have to duplicate the function of the code all across the place... And since we want to be `no-std` this seems to be the best option.

Reply:
Ok then.